p2p/server/msgserver.go

```diff
@@ -49,29 +50,73 @@ func extractPayload(m Message) []byte {
 	return data.Payload
 }
 
-// Item is queue entry used to match responds to sent requests.
-type Item struct {
+// item is queue entry used to match responds to sent requests.
+type item struct {
 	id        uint64
 	timestamp time.Time
 }
 
-// ResponseHandlers contains handlers for received response handlers
-type ResponseHandlers struct {
+type responseHandlers struct {
 	okCallback   func(msg []byte)
 	failCallBack func(err error)
 }
 
+// ErrShuttingDown is returned to the peer when the node is shutting down
+var ErrShuttingDown = errors.New("node is shutting down")
+
+// ErrBadRequest is returned to the peer upon failure to parse the request
+var ErrBadRequest = errors.New("unable to parse request")
+
+var errRequestTimeout = errors.New("request timed out")
```

Comment:
better to make it public so that requestor can identify if request timed out with errors.Is

Reply:
done